mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/NameUtils.java

```diff
@@ -289,13 +290,34 @@ public static String defaultNamespace(String extensionName) {
       throw new IllegalArgumentException("No Namespace can be created from an empty name");
     }
 
-    String namespace = deleteWhitespace(extensionName);
+    String namespace = extensionName.trim();
     namespace = removeEndIgnoreCase(namespace, "extension");
     namespace = removeEndIgnoreCase(namespace, "connector");
     namespace = removeEndIgnoreCase(namespace, "module");
+
+    if (namespace.contains(SPACE)) {
+      namespace = toProperCase(namespace);
```

Comment:
what's the point of this method when `hypenize()` is going to end up lower casing everything anyway?

Reply:
Hi @marianogonzalez,

This issue is related with the Custom Properties Provider Extension name. The deploy fails when a default namespace is made up of words in camelcase separated by spaces, and where one of them also has another uppercase character in between, for example `AbCd Properties Provider`. What's happening is that the `hyphenize()` separate the string in words taking into account the upper case letters, which it give us the following transformation: `Ab Cd Properties Provider` -> `ab-cd-properties-provider` that causes a failure deployment because that namespace doesn't match with the one that it is generated by the Config `abcd-properties-provider`.

We decided to make the fix when the namespace is processed in this side of the code, but now we are concern if we should investigate how the Config is created and how the namespace is processed to understand if it is appropriate or not, to leave this code as it is and make a fix when the Config is created.

What do you think about it?

Thank you!